### PR TITLE
task/tup-555 order news article categories

### PR DESCRIPTION
**Pull request by R-Tomas-Gonzalez**
## Overview

Orders the news article categories.

* **Check Notes**

## Related

https://tacc-main.atlassian.net/browse/TUP-555

## Changes

- Creates three different lists to check... `'press-release, feature-story'`, `'multimedia, podcast'`, and `'press-release, feature-story, multimedia, podcast'`
- Logic: The `if` statement checks the slug against the strings in the first two, and checks if it is not in the last
-- The logic above will order the lists as such
1. press-release or feature-story
2. multimedia or podcast
3. any others any order

## Testing

1. Create categories in Core-CMS image
2. Add said categories (with some random ones) to a news story
3. Test the order

## UI

Screenshot 2023-12-19 at 7 01 07 PM

## Notes
~~I came away with an issue. @wesleyboar Please let me know if this is an issue or if we can troubleshoot this together.
_Issue:_ Although the categories are in the order we'd like them to be, they class numbers are **not** correlated with their order on the page. Unsure of a solution at the moment. But working to fix.~~

**Edit**: This is no longer issue as of [05d4e7f](https://github.com/TACC/Core-CMS/pull/776/commits/05d4e7f39a02ca78c304b209f9b0bf6f25a104d7)
Screenshot 2023-12-19 at 7 03 41 PM

### CODE REVIEW

**Review by wesleyboar — CHANGES_REQUESTED** [review]
**Good!** This is an excellent base upon which to refine this feature.

1. <details><summary>Move this new logic to its own template, then include that template.</summary>

    This will isolate complex logic specific to one small aspect of the template.
    
    <sup>Examples: I did this with [`includes/blog_cats.html`](https://github.com/TACC/Core-CMS/blob/v4.3.0/taccsite_cms/templates/djangocms_blog/includes/blog_cats.html) (included by [`base.html`](https://github.com/TACC/Core-CMS/blob/v4.3.0/taccsite_cms/templates/djangocms_blog/base.html#L29)). (I should do this to [dates](https://github.com/TACC/Core-CMS/blob/v4.3.0/taccsite_cms/templates/djangocms_blog/includes/blog_meta.html#L17-L42).)</sup>

    </details>
2. <details><summary><strong>Suggestion</strong>: Move this logic into Python (so it easier to grok and improve).</summary>

    You can move logic to Python via a [template tag or filter (Django docs)](https://docs.djangoproject.com/en/4.2/howto/custom-template-tags/).
    
    <sup>Example: An existing template tag [`blog_post_use_custom_media.py`](https://github.com/TACC/Core-CMS/blob/v4.3.0/taccsite_cms/templatetags/blog_post_use_custom_media.py) that works with News categories.</sup>

    </details>
 3. <details><summary>Consider how to reduce duplicate logic. (Probably easier w/ Python code.)</summary>

    I see 3 for loops, and 3 `<li>`'s. Hopefully, this can be done with less of each.

    </details>
4. <details><summary><strong>Bonus</strong>: Consider whether and how to integrate with Core-CMS <code>settings</code>.</summary>

    Other CMS's would have different categories, and maybe a different order.
    
    <sup>Example: [Setting `TACC_SOCIAL_SHARE_PLATFORMS`](https://github.com/TACC/Core-CMS/blob/v4.3.0/taccsite_cms/settings.py#L293) is an array used by [`share_on_social.html`](https://github.com/TACC/Core-CMS/blob/v4.3.0/taccsite_cms/templates/share_on_social.html).</sup>

    </details>

**Comment by R-Tomas-Gonzalez** [comment]
[7c52f1c](https://github.com/TACC/Core-CMS/pull/776/commits/7c52f1cc1c6025154bfc24ec30ea994e75935f54)
Accomplishes the following:

1. <details><summary><s>Move this new logic to its own template, then include that template.</s></summary>

    This will isolate complex logic specific to one small aspect of the template.
    
    <sup>Examples: I did this with [`includes/blog_cats.html`](https://github.com/TACC/Core-CMS/blob/v4.3.0/taccsite_cms/templates/djangocms_blog/includes/blog_cats.html) (included by [`base.html`](https://github.com/TACC/Core-CMS/blob/v4.3.0/taccsite_cms/templates/djangocms_blog/base.html#L29)). (I should do this to [dates](https://github.com/TACC/Core-CMS/blob/v4.3.0/taccsite_cms/templates/djangocms_blog/includes/blog_meta.html#L17-L42).)</sup>

    </details>
2. <details><summary><s><strong>Suggestion</strong>: Move this logic into Python (so it easier to grok and improve).</s></summary>

    You can move logic to Python via a [template tag or filter (Django docs)](https://docs.djangoproject.com/en/4.2/howto/custom-template-tags/).
    
    <sup>Example: An existing template tag [`blog_post_use_custom_media.py`](https://github.com/TACC/Core-CMS/blob/v4.3.0/taccsite_cms/templatetags/blog_post_use_custom_media.py) that works with News categories.</sup>

    </details>
 3. <details><summary><s>Consider how to reduce duplicate logic. (Probably easier w/ Python code.)</s></summary>

    I see 3 for loops, and 3 `<li>`'s. Hopefully, this can be done with less of each.

    </details>
4. <details><summary><strong>Bonus</strong>: Consider whether and how to integrate with Core-CMS <code>settings</code>.</summary>

    Other CMS's would have different categories, and maybe a different order.
    
    <sup>Example: [Setting `TACC_SOCIAL_SHARE_PLATFORMS`](https://github.com/TACC/Core-CMS/blob/v4.3.0/taccsite_cms/settings.py#L293) is an array used by [`share_on_social.html`](https://github.com/TACC/Core-CMS/blob/v4.3.0/taccsite_cms/templates/share_on_social.html).</sup>

    </details>

File: taccsite_cms/templates/djangocms_blog/includes/blog_meta.html [hunk 1]
```diff
@@ -61,21 +61,7 @@
...
-                <li class="category_{{ forloop.counter }}"><a href="{% url 'djangocms_blog:posts-category' category=category.slug %}" class="blog-category-{{ category.slug }}">{{ category.name }}</a></li>
-                {# /TACC #}
-                {# /TACC #}
-                {# /TACC #}
-            {% endif %}
-        {% endfor %}
-    {% else %}
-      <li class="category_none">{% trans "Uncategorized" %}</li>
-    {% endif %}
+{% include './blog_cats_organized.html' %}
```
Comment (wesleyboar) [inline taccsite_cms/templates/djangocms_blog/includes/blog_meta.html:64]: I promise to stop commenting on differences from original template… after [TUP-656](https://tacc-main.atlassian.net/browse/TUP-656) or after upgrading to [djangocms-blog v2](https://github.com/nephila/djangocms-blog/releases).

```suggestion
    {# TACC (nephila/djangocms-blog#730, no commas, rename class, order cats): #}
    {% include './blog_cats_organized.html' %}
    {# /TACC #}
```

File: taccsite_cms/templatetags/blog_cats_organized.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+from django import template
+from django.conf import settings
+
+TACC_BLOG_CATEGORY_ORDER = settings.TACC_BLOG_CATEGORY_ORDER
+register = template.Library()
```
Comment (wesleyboar) [inline taccsite_cms/templatetags/blog_cats_organized.py:5]: This has been done previously in Core-CMS by passing settings through template to the templatetag (as in https://github.com/TACC/Core-CMS/pull/594), but I'll review this alternative after the holidays. I'm not opposed; I just wanna think on it.

**Review by wesleyboar — COMMENTED** [review]
Very nice.

File: taccsite_cms/templatetags/blog_cats_organized.py (hunk 2 above)
Comment (wesleyboar) [reply taccsite_cms/templatetags/blog_cats_organized.py:5]: This is a sensible way to access the setting, and better than how I would have done it.

@taoteg, I learned something. If you already knew it, just ignore me. The `SETTINGS_EXPORT` is not required for explicitly using a setting in the context of a template; a templatetag can import one instead. 🙂

@R-Tomas-Gonzalez, Do you think (outside this PR) you could change `TACC_BLOG_CUSTOM_MEDIA_POST_CATEGORY` to be used this same way?

**Review by wesleyboar — APPROVED** [review]
Regarding the issue about category numbers not being in order:
1. I don't mind. We don't use those class names.[^1]
2. [I failed to replicate the issue. They are in order for me.](https://github.com/TACC/Core-CMS/assets/62723358/fe047ca6-0858-40af-9f02-b4e45d3a3856)


[^1]: Most of the blog/news class names came from a reference template. I might even replace them with TACC class names.

File: taccsite_cms/templatetags/blog_cats_organized.py (hunk 2 above)
Comment (R-Tomas-Gonzalez) [reply taccsite_cms/templatetags/blog_cats_organized.py:5]: Will do.

**Pull request merged**